app/grandchallenge/reader_studies/serializers.py

```diff
@@ -31,6 +32,14 @@ class Meta:
             "reader_study",
             "required",
         )
+        swagger_schema_fields = swagger_schema_fields_for_charfield(
+            ("answer_type", "form_direction", "image_port"),
+            (
+                Question.ANSWER_TYPE_CHOICES,
+                Question.DIRECTION_CHOICES,
+                Question.IMAGE_PORT_CHOICES,
+            ),
```

Comment:
This syntax could be improved, always better to use keyword only arguments and if your passing data together it is better to use a dictionary rather than 2 lists that can get out of step. Could you do something like:

```
swagger_schema_fields_for_charfield(fields=["answer_type", "form_direction", "image_port"], model=Question)
```

and then within that function:

```
def swagger_schema_fields_for_charfield(*, model, fields):
   ...
   fields_with_choices = {f: model._meta.field(f).choices for f in fields}
```

If you can find a way to do this without passing the model (DRY) then even better.

Reply:
These have been changed as discussed during our meeting.